src/BlockEntities/ChestEntity.cpp

```diff
@@ -219,32 +218,16 @@ void cChestEntity::DestroyWindow()
 
 
 
-class cFindSittingCat :
-	public cEntityCallback
-{
-	virtual bool Item(cEntity * a_Entity) override
-	{
-		return (
-			(a_Entity->GetEntityType() == cEntity::etMonster) &&
-			(static_cast<cMonster *>(a_Entity)->GetMobType() == eMonsterType::mtOcelot) &&
-			(static_cast<cOcelot *>(a_Entity)->IsSitting())
-		);
-	}
-};
-
-
-
-
-
 bool cChestEntity::IsBlocked()
 {
-	cFindSittingCat FindSittingCat;
 	return (
-		(GetPosY() >= cChunkDef::Height - 1) ||
-		!cBlockInfo::IsTransparent(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ())) ||
+		(GetPosY() < cChunkDef::Height - 1) &&
 		(
-			(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ()) == E_BLOCK_AIR) &&
-			!GetWorld()->ForEachEntityInBox(cBoundingBox(Vector3d(GetPosX(), GetPosY() + 1, GetPosZ()), 1, 1), FindSittingCat)
+			!cBlockInfo::IsTransparent(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ())) ||
+			(
+				(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ()) == E_BLOCK_AIR) &&
+				!cOcelot::IsCatSittingOnBlock(GetWorld(), Vector3d(GetPos()))
+			)
```

Comment:
I think it might be easier to understand if you broke this return statement into several `if` statements.
```cpp
// Check if there is a solid block above the chest:
if (GetPosY() < cChunkDef::height - 1)
{
  // <check block solidity>
}

// Check if there's a cat sitting on top:
if (cOcelot::IsCatSittingOnBlock(...))
{
  return true;
}

// Not blocked
return false;
```
(Note that I'm assuming that a cat sitting on top of a chest is not affected by the block type above the chest, and the check still works if the chest is at the top of the world, with the cat sitting on it being above the world)